src/CodeGen_Hexagon.cpp

```diff
@@ -501,22 +501,27 @@ llvm::Function *CodeGen_Hexagon::define_hvx_intrinsic(llvm::Function *intrin, Ty
     return wrapper;
 }
 
+Value *CodeGen_Hexagon::create_bitcast(Value *v, llvm::Type *ty) {
+    if (BitCastInst *c = dyn_cast<BitCastInst>(v)) {
+        return create_bitcast(c->getOperand(0), ty);
```

Comment:
This is strange. Isn't LLVM able to cancel out redundant bitcasts? Is this leading to sub-optimal code?